{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/validation/model_validation_input.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+from enum import Enum
+from mlflow.models import MetricThreshold, make_metric
+
+
+class RunMode(Enum):
+    ENABLED = 1
+    DRY_RUN = 2
+    DISABLED = 3
+
+
+# The `run_mode` defines whether model validation is enabled or not. It can be one of the three values:
+# - `DISABLED` : Do not run the model validation notebook.
+# - `DRY_RUN`  : Run the model validation notebook. Ignore failed model validation rules and proceed to move model to Production stage.
+# - `ENABLED`  : Run the model validation notebook. Move model to Production stage only if all model validation rules are passing.
+# TODO(required)
+def get_run_mode():
+    return RunMode.DISABLED
+
+
+# Whether to load the current registered "Production" stage model as baseline. A version with "Production" stage must
+# exist for the model.
+# Baseline model is a requirement for relative change and absolute change validation rules.
+# TODO(required)
+def enable_baseline_comparison():
+    return False
+
+
+# model validation data input for prod workspace. A Pandas DataFrame or Spark DataFrame, containing evaluation features and labels.
+# Please refer to data parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required)
+def get_prod_workspace_validation_input(spark):
+    return spark.sql(
+        "SELECT * FROM delta.`dbfs:/databricks-datasets/nyctaxi-with-zipcodes/subsampled`"
+    )
+
+
+# model validation data input for staging workspace. A Pandas DataFrame or Spark DataFrame, containing evaluation features and labels.
+# Please refer to data parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required)
+def get_staging_workspace_validation_input(spark):
+    return spark.sql(
+        "SELECT * FROM delta.`dbfs:/databricks-datasets/nyctaxi-with-zipcodes/subsampled`"
+    )
+
+
+# A string describing the model type. The model type can be either "regressor" and "classifier".
+# Please refer to model_type parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required) : model_type
+def get_model_type():
+    return "regressor"
+
+
+# The string name of a column from data that contains evaluation labels.
+# Please refer to targets parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required) : targets
+def get_targets():
+    return "mean_squared_error"
+
+
+# Custom metrics to be included. Set it to None if custom metrics are not needed.
+# Please refer to custom_metrics parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : define custom metric function to be included in custom_metrics.
+def squared_diff_plus_one(eval_df, _builtin_metrics):
```

Comment:
Why don't we fold this logic inside get_custom_metrics()? Then it becomes simpler in terms of understanding exactly what the inputs are to the notebook.